src/daemon/control.rs

```diff
@@ -1699,4 +1707,492 @@ mod test {
+        check_signature(
+            &revaultd.secp_ctx,
+            &another_tx,
+            public_key.clone(),
+            &sig,
+            SigHashType::All,
+        )
+        .unwrap_err();
+
+        // Checking against a completely different signature
+        let hex = Vec::<u8>::from_hex("304402203d81027376dbad3ed5632ceb3769b2020d0eae90c0f9543828048c3eeb1b825a02201b988847ee7d18930644eab0a0906799fdc68a0fb639270edf3ed0889b597fee").unwrap();
+        let another_sig = secp256k1::Signature::from_der(&hex).unwrap();
+        check_signature(
+            &revaultd.secp_ctx,
+            &tx,
+            public_key,
+            &another_sig,
+            SigHashType::All,
+        )
+        .unwrap_err();
+    }
+
+    #[test]
+    fn test_check_revocation_signatures() {
+        let datadir = test_datadir();
+        let revaultd = dummy_revaultd(datadir, UserRole::ManagerStakeholder);
+
+        // We need a ctx that can sign as well (revaultd context is verify only)
+        let ctx = secp256k1::Secp256k1::new();
+
+        let (secret_key, _, public_key) =
+            create_keys(&ctx, &[1; secp256k1::constants::SECRET_KEY_SIZE]);
+
+        // Let's forge a valid signature
+        let tx = UnvaultEmergencyTransaction::from_psbt_str("cHNidP8BAF4CAAAAAbmw9RR44LLNO5aKs0SOdUDW4aJgM9indHt2KSEVkRNBAAAAAAD9////AaQvhEcAAAAAIgAgy7Co1PHzwoce0hHQR5RHMS72lSZudTF3bYrNgqLbkDYAAAAAAAEBK9BxhEcAAAAAIgAgMJBZ5AwbSGM9P3Q44qxIeXv5J4UXLnhwdfblfBLn2voBAwSBAAAAAQWoIQL5vFDdmMV/P4SpzIWhDMbHKHMGlMwntZxaWtwUXd9KvKxRh2R2qRTtnZLjf14tI1q08+ZyoIEpuuMqWYisa3apFJKFWLx/I+YKyIXcNmwC0yw69uN9iKxsk1KHZ1IhAw9kuSKu4v1ZfxBLxss7Zw8cosbEmxBxoabAEFddlP5aIQKr5HWxmew9YvpXb67hajNP24b/sm3Odb7Ouq7fMorD/lKvU7JoAAA=").unwrap();
+        let psbt = tx.inner_tx();
+        let mut cache = SigHashCache::new(&psbt.global.unsigned_tx);
+        let prev_value = psbt.inputs[0].witness_utxo.as_ref().unwrap().value;
+        let script_code = psbt.inputs[0].witness_script.as_ref().unwrap();
+        let sighash = cache.signature_hash(
+            0,
+            &script_code,
+            prev_value,
+            SigHashType::AllPlusAnyoneCanPay,
+        );
+        let sighash = secp256k1::Message::from_slice(&sighash).unwrap();
+        let mut sig = ctx.sign(&sighash, &secret_key).serialize_der().to_vec();
+        sig.push(SigHashType::AllPlusAnyoneCanPay as u8);
+        let mut sigs = BTreeMap::new();
+        sigs.insert(public_key, sig.clone());
+
+        // Happy path: everything works :)
+        check_revocation_signatures(&revaultd.secp_ctx, &tx, &sigs).unwrap();
+
+        // Now, onto with the sad paths...
+
+        // Signature is empty? Wtf?
+        let mut wrong_sigs = BTreeMap::new();
+        wrong_sigs.insert(public_key, vec![]);
+        check_revocation_signatures(&revaultd.secp_ctx, &tx, &wrong_sigs).unwrap_err();
```

Comment:
I usually prefer to assert specific errors. This makes pickier tests which is sometimes annoying but you are grateful once it's triggered because you changed the logic and it's not failing for the right reason anymore :)

Reply:
I agree, the trouble was that I can't derive `PartialEq` on `RpcControlError`:

```
error[E0369]: binary operation `==` cannot be applied to type `revault_tx::Error`
   --> src/daemon/control.rs:143:8
    |
143 |     Tx(revault_tx::Error),
    |        ^^^^^^^^^^^^^^^^^
    |
    = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Can you think of another solution to avoid the unwrap_err but also avoid changing revault_tx?

Reply:
Yeah, i usually do the check-string-repr hack: `unwrap_err().to_string().contains("blabla")`.

Reply:
Should be fixed, I also fixed the RPC communication routines tests in another commit :)

Reply:
Awesome